GVRf/Framework/src/org/gearvrf/scene_objects/GVRTextViewSceneObject.java

```diff
@@ -0,0 +1,310 @@
+import android.widget.TextView;
+
+public class GVRTextViewSceneObject extends GVRSceneObject implements
+        GVRDrawFrameListener {
+    private static final int REFRESH_INTERVAL = 30; // frames
+
+    private final Surface mSurface;
+    private final SurfaceTexture mSurfaceTexture;
+    private final LinearLayout mTextViewContainer;
+    private final TextView mTextView;
+    private static int DEFAULT_WIDTH = 2000;
+    private static int DEFAULT_HEIGHT = 1000;
+    private static float DEFAULT_QUAD_WIDTH = 2.0f;
+    private static float DEFAULT_QUAD_HEIGHT = 2.0f;
+    private int mCount = 0;
+
+    /**
+     * Shows a {@link TextView} on a {@linkplain GVRSceneObject scene object}.
+     * 
+     * @param gvrContext
+     *            current {@link GVRContext}
+     * @param mesh
+     *            a {@link GVRMesh} - see
+     *            {@link GVRContext#loadMesh(org.gearvrf.GVRAndroidResource)}
+     *            and {@link GVRContext#createQuad(float, float)}
+     * @param gvrActivity
+     *            a {@link GVRActivity}
+     * @param viewWidth
+     *            Width of the {@link TextView}
+     * @param viewHeight
+     *            Height of the {@link TextView}
+     * @param text
+     *            {@link String} to show on the textView
+     */
+
+    public GVRTextViewSceneObject(GVRContext gvrContext, GVRMesh mesh,
+            GVRActivity gvrActivity, int viewWidth, int viewHeight, String text) {
+        super(gvrContext, mesh);
+        mTextView = new TextView(gvrActivity);
+        mTextView.setLayoutParams(new LayoutParams(viewWidth, viewHeight));
+        mTextView.measure(viewWidth, viewHeight);
+        mTextView.setBackgroundColor(Color.TRANSPARENT);
+        mTextView.setText(text);
+        mTextView.setVisibility(View.VISIBLE);
+        mTextViewContainer = new LinearLayout(gvrActivity);
+        mTextViewContainer.addView(mTextView);
+        mTextViewContainer.measure(viewWidth, viewHeight);
+        mTextViewContainer.layout(0, 0, viewWidth, viewHeight);
+        mTextViewContainer.setVisibility(View.VISIBLE);
+        gvrContext.registerDrawFrameListener(this);
+        GVRTexture texture = new GVRExternalTexture(gvrContext);
+        GVRMaterial material = new GVRMaterial(gvrContext, GVRShaderType.OES.ID);
+        material.setMainTexture(texture);
+        getRenderData().setMaterial(material);
+
+        mSurfaceTexture = new SurfaceTexture(texture.getId());
+        mSurface = new Surface(mSurfaceTexture);
+        mSurfaceTexture.setDefaultBufferSize(mTextViewContainer.getWidth(),
+                mTextViewContainer.getHeight());
+    }
+
+    /**
+     * Shows a {@link TextView} on a {@linkplain GVRSceneObject scene object}.
+     * 
+     * @param gvrContext
+     *            current {@link GVRContext}
+     * @param gvrActivity
+     *            a {@link GVRActivity}
+     * @param width
+     *            width for createQuad to get a new {@link GVRMesh}
```

Comment:
This is true, strictly speaking, but probably not quite as helpful as it could be: There's potential for confusion between the __GVRf size__ and the __Android size__.